EventEnrichment/src/main/java/com/amazonaws/utils/S3LoadData.java

```diff
@@ -0,0 +1,169 @@
+                         * An End Event informs that the request has finished successfully.
+                         */
+                        @Override
+                        public void visit(SelectObjectContentEvent.EndEvent event)
+                        {
+                            isResultComplete.set(true);
+                        }
+                    }
+            );
+
+            location = new BufferedReader(
+                    new InputStreamReader(resultInputStream, StandardCharsets.UTF_8))
+                    .lines()
+                    .collect(Collectors.joining("\n"));
+
+            List<String> roleLocations = Arrays.stream(location
+                            .split("\n"))
+                            .collect(Collectors.toList());
+
+            for (String roleLocation: roleLocations) {
+                String role = roleLocation.split(",")[0];
+                String building = roleLocation.split(",")[1];
+                locationMap.put(role, new Location(role, building));
+            }
+
+            //System.out.println(locationMap);
+
+        }
+
+        /*
+         * The End Event indicates all matching records have been transmitted.
+         * If the End Event is not received, the results may be incomplete.
+         */
+        if (!isResultComplete.get()) {
+            throw new Exception("S3 Select request was incomplete as End Event was not received.");
+        }
+
+        return locationMap;
+    }
+
+    private SelectObjectContentRequest generateBaseCSVRequest(String bucket, String key, String query) {
+        SelectObjectContentRequest request = new SelectObjectContentRequest();
+        request.setBucketName(bucket);
+        request.setKey(key);
+        request.setExpression(query);
+        request.setExpressionType(ExpressionType.SQL);
+
+        InputSerialization inputSerialization = new InputSerialization();
+        inputSerialization.setCsv(new CSVInput().withFileHeaderInfo(FileHeaderInfo.USE));
+        inputSerialization.setCompressionType(CompressionType.NONE);
+        request.setInputSerialization(inputSerialization);
+
+        OutputSerialization outputSerialization = new OutputSerialization();
+        outputSerialization.setCsv(new CSVOutput());
+        request.setOutputSerialization(outputSerialization);
+
+        return request;
+    }
+
+    public Location loadReferenceLocationData(String key) throws Exception {
+        System.out.println();
+        System.out.println("***************************************");
+        System.out.println("Loading Location Reference Data for key: " +  key);
+
+        Location location = new Location(key, getLocation(key));
+
+        System.out.println(location);
+        System.out.println("***************************************");
+        System.out.println();
+
+        return location;
+    }
+
+    public Map<String, Location> loadReferenceLocationData() throws Exception {
+        System.out.println();
```

Comment:
since this will realistically be called with each message, should we remove these print statements?

Reply:
Replaced this with logger.